packages/contracts-bedrock/src/safe/DeputyPauseModule.sol

```diff
@@ -210,4 +215,26 @@ contract DeputyPauseModule is ISemver, EIP712 {
         // Emit the DeputySet event.
         emit DeputySet(_deputy);
     }
+
+    /// @notice Internal function to set the DeputyGuardianModule.
+    /// @param _deputyGuardianModule DeputyGuardianModule address.
+    function _setDeputyGuardianModule(IDeputyGuardianModule _deputyGuardianModule) internal {
+        // Cannot set the DeputyGuardianModule to be the Foundation Safe or there's some potential
+        // to cause the Foundation Safe to trigger itself if there's some important function that
+        // happens to have the "pause()" selector.
+        if (address(_deputyGuardianModule) == address(FOUNDATION_SAFE)) {
+            revert DeputyPauseModule_InvalidDeputyGuardianModule();
+        }
+
+        // Make sure that the DeputyGuardianModule actually has code.
+        if (address(_deputyGuardianModule).code.length == 0) {
+            revert DeputyPauseModule_InvalidDeputyGuardianModule();
+        }
+
+        // Set the DeputyGuardianModule.
+        deputyGuardianModule = _deputyGuardianModule;
+
+        // Emit the DeputyGuardianModuleSet event.
+        emit DeputyGuardianModuleSet(_deputyGuardianModule);
+    }
```

Comment:
since we're doing all this validation of the DGM, should we also confirm that [these values are](https://github.com/ethereum-optimism/optimism/blob/bc02e90ea8f90fa9259d83325f8dc7b633fcbc1a/packages/contracts-bedrock/src/safe/DeputyGuardianModule.sol#L42-L48) properly configured?

Reply:
We elected not to do that